<launch>

	<!-- USBポート番号の定義 -->
	
    <arg name="canusb_port"        value="/dev/ttyUSB0"/>
	<arg name="frontLRF_port"      value="/dev/ttyACM1"/>
    <arg name="front_tiltLRF_port" value="/dev/ttyACM3"/>
    <arg name="backLRF_port"       value="/dev/ttyACM0"/>
    <arg name="imu_port"           value="/dev/ttyACM2"/>

	<!-- 各種センサへの接続 -->
    
    <node pkg="ultimate_seniorcar" type="canusb_connecter.py" name="can_usbconnecter">
        <param name="canusb_port" value="$(arg canusb_port)"/>
    </node>
    
    <include file="$(find ultimate_seniorcar)/launch_xml/hokuyo_lidar.xml"> 
    	<arg name="frontLRF_port"       value="$(arg frontLRF_port)"/> 
        <arg name="front_tiltLRF_port"  value="$(arg front_tiltLRF_port)"/> 
        <arg name="backLRF_port"        value="$(arg backLRF_port)"/> 
    </include>

    <include file="$(find ultimate_seniorcar)/launch_xml/imu_and_pose_estimation.xml">
        <arg name="imu_port" value="$(arg imu_port)"/>
    </include>

    <!-- 各種ノード立ち上げ -->
    <node pkg="ultimate_seniorcar" type="seniorcar_odometry.py" name="seniorcar_odometry" />
    <node pkg="urg_process" type="base_link_to_odom.py" name="base_link_to_odom">
        <param name="odom_topic" value="seniorcar_odometry"/>
    </node>
    <!--<include file="$(find ultimate_seniorcar)/launch_xml/gmapping.xml" />-->

    <!-- おまけ -->
    <!--<node pkg="rviz" type="rviz" name="rviz" args="-d $(find joystic_wheelchair)/launch_xml/costmap_and_laser.rviz"/>-->

</launch>



<!-- rosbagで必要なトピックだけをレコード -->
<!-- rosrun rosbag record scan scan_tilt encoded_odom imu/data imu/data_raw imu/mag tf -o /home/ishikawa/bagfiles --> 